ui-v2/app/services/repository.js

```diff
@@ -52,6 +57,7 @@ export default Service.extend({
     });
   },
   invalidate: function() {
+    // TODO: This should probably return a Promise
```

Comment:
I don't think the underlying methods do (yet) - there was some talk of changing the ember data api to do this, but I don't remember where it landed / what progress there was.

Reply:
Yeah for sure, this was more a TODO to make my Repository class methods all use a more consistent Promise based interface. Most of the methods return Promises, so I would like these ones to also.